src/main/java/com/diffplug/gradle/eclipserunner/EquinoxLauncher.java

```diff
@@ -107,7 +110,12 @@ private File getPluginRequireSingle(String name) {
 
 	/** Sets the application arguments which will be passed to the runtime. */
 	public EquinoxLauncher setArgs(List<String> args) {
-		this.args = ImmutableList.copyOf(args);
+		// Filter arguments that where meant for the original launcher (eclipsec.exe)
+		List<String> filteredArgs = args.stream()
+				.filter(Objects::nonNull)
+				.filter(arg -> !arg.startsWith("--launcher"))
+				.collect(toList());
+		this.args = ImmutableList.copyOf(filteredArgs);
 		return this;
```

Comment:
What is this filter for?  If I pass `--launcher.someFlag`, it will be silently discarded, which will be quite hard to debug.

Reply:
I needed that for the P2BootstrapInstallationTest to pass, since unknown arguments will lead to an error (See log output at the end of this post). As i understood it, EquinoxLauncher  is an alternate way to startup equinox, without having to use the native Eclipse launchers? The "--launcher" arguments are usually meant for the eclipse.exe or eclipsec.exe launchers and not passed further to Equinox. Please correct me, if my understanding is wrong. I would be glad for any better solution.


Installation failed.
!SESSION 2018-02-02 20:24:20.472 -----------------------------------------------
eclipse.buildId=unknown
java.version=1.8.0_112
java.vendor=Oracle Corporation
BootLoader constants: OS=macosx, ARCH=x86_64, WS=cocoa, NL=de_DE
Framework arguments:  --launcher.suppressErrors -application org.eclipse.equinox.p2.director -repository http://download.eclipse.org/eclipse/updates/4.5/R-4.5.2-201602121500/ -installIU org.eclipse.core.runtime -profile profile -destination file:///var/folders/4d/z5fm5n5x5gsdzwswfrfd7b100000gn/T/junit191409443383839977/installed
Command-line arguments:  --launcher.suppressErrors -application org.eclipse.equinox.p2.director -clean -consolelog -repository http://download.eclipse.org/eclipse/updates/4.5/R-4.5.2-201602121500/ -installIU org.eclipse.core.runtime -profile profile -destination file:///var/folders/4d/z5fm5n5x5gsdzwswfrfd7b100000gn/T/junit191409443383839977/installed

!ENTRY org.eclipse.equinox.p2.core 4 0 2018-02-02 20:24:20.474
!MESSAGE Unknown option --launcher.suppressErrors. Use -help for a list of known options.

Reply:
Ahhhh, I see.  You are correct about EqunoxLauncher.

I think a safer bet here would be to filter `--launcher.suppressErrors` only.  There are [quite a few `--launcher` flags](https://help.eclipse.org/neon/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fmisc%2Fruntime-options.html), and it may be that some of them are supported by EquinoxLauncher.

Reply:
Allright. I changed that